WalletWasabi/Blockchain/TransactionBuilding/ChangelessTransactionCoinSelector.cs

```diff
@@ -75,7 +76,17 @@ internal static bool TryGetCoins(SelectionStrategy strategy, long target, Dictio
 
 		BranchAndBound branchAndBound = new();
 
-		var foundExactMatch = branchAndBound.TryGetMatch(strategy, out List<long>? solution, cancellationToken);
+		bool foundExactMatch = false;
+		List<long>? solution = null;
+
+		try
```

Comment:
My understanding here is that we can compute a "LESS" suggestion and a "MORE" suggestion can time out and that in turn does not affect "LESS" suggestion. Is that correct?

Reply:
If I'm right, then your PR actually improves the feature and it is not only a refactoring. Right?

Reply:
> My understanding here is that we can compute a "LESS" suggestion and a "MORE" suggestion can time out and that in turn does not affect "LESS" suggestion. Is that correct?

No, they use the same `cancellation token`, so they will time out at the same time.

Do you have a scenario in mind where we should have two separate cancellation token, so we can cancel them separately?

IMO we don't need that because: 
- If the User navigates away from the `Transaction Preview Dialog`, we want to cancel both of them.
- If the User changes pockets, we want to cancel both of them and start the new search with the newly selected pocket.
- If the User changes fee, we want to cancel both of them and start it again with the new `FeeRate`.
- If they run for more than 15 seconds, we want to cancel both of them and show the closest matches.
- If one of them finishes faster, we show that suggestion while the other is still searching. They don't block each other.

Reply:
I don't know precisely but what I'm interested in is whether user experience for the following scenarios is good:

* "More" suggestion is computed fast and "Less" takes forever
* "Less" suggestion is computed fast and "More" takes forever
* Both "Less" and "More" suggestions take forever to compute
* Both "Less" and "More" suggestions are computed fast

I don't have a specific suggestion, I'm just trying to make sure we provide good user experience for these scenarios as they will be common.

Also given that these privacy suggestions is a really good feature, I think it makes sense to spend a little bit of time to make sure it behaves great.

Reply:
> I think it makes sense to spend a little bit of time to make sure it behaves great.

I couldn't agree more.

AFAIK:

> "More" suggestion is computed fast and "Less" takes forever

We show the "More" when it's computed, then after 15 second we show the "Less".

> "Less" suggestion is computed fast and "More" takes forever

Same as above but dwitch the "More" and the "Less"

> Both "Less" and "More" suggestions take forever to compute

After 15 second we show them.

> Both "Less" and "More" suggestions are computed fast (happy path)

We show them as they finish, they don't wait for each other.

But these scenarios only valid if the found solution is `target * 0.75 <= solution <= target * 1.25`, so they land between our percentage "tolerance".

Reply:
Sounds great!